### ISSUE TITLE
SameExecutorCompletionStage - Java 9 upgrade

### ISSUE THREAD

**Issue by billoneil** [jOOQ/jOOL#298]
Looks like Java 9 supports overriding the default executor in `CompletableFuture`. http://download.java.net/java/jdk9/docs/api/java/util/concurrent/CompletableFuture.html#defaultExecutor--

This could simplify the implementation of `SameExecutorCompletionStage` in the future but not until support for java8 is dropped.

Does this kill the need for this class entirely? It would be best to remove it before the release if so. I think it might be nice to have for Java 8 though.

**Comment by lukaseder** [comment]
Hmm, interesting discovery, thanks for sharing. I hadn't looked at these Java 9 improvements yet.

I think it's definitely worth mentioning in the `Async` Javadoc, but these improvements are only valid for `CompletableFuture` (not `CompletionStage`), whose design I still don't fully agree with. So I think we can keep the implementation you've added, given that Java 9 will take quite a while to be adopted.

I'll rename this issue to be the task about adding a comment to the Javadoc...
